gprofiler/python.py

```diff
@@ -256,6 +256,12 @@ def snapshot(self) -> Mapping[int, Mapping[str, int]]:
         collapsed_path = self._dump()
         collapsed_text = collapsed_path.read_text()
         collapsed_path.unlink()
+        # PyPerf outputs sampling & error counters every interval, print them.
+        # also, makes sure its output pipe doesn't fill up.
+        # using read1() which performs just a single read() call and doesn't read until EOF
+        # (unliked Popen.communicate())
+        assert self.process is not None
+        logger.debug(f"PyPerf output: {self.process.stderr.read1()}")
```

Comment:
This should be in `_dump` in case an exception is thrown between it and this line.

Reply:
Where can an exception be thrown? The only thing between is reading a file that we know exists, and unlinking it

Reply:
The read show be performed after the file is written because we want to see the latest output

Reply:
After `_wait_for_output_file` the file is written.

Reply:
IK, so between `_wait_for_output_file` to this current point, which exception can be raised?

> The only thing between is reading a file that we know exists, and unlinking it

Reply:
Eh, best to have the output logged asap

Reply:
Well since it's just the read of a file between, I don't think it matters much. But okay, will move